geemap/map_widgets.py

```diff
@@ -1022,53 +1022,56 @@ def _observe_visible(self, change: Dict[str, Any]) -> None:
 
 
 @Theme.apply
-class Basemap(ipywidgets.HBox):
+class BasemapSelector(anywidget.AnyWidget):
     """Widget for selecting a basemap."""
 
+    _esm = pathlib.Path(__file__).parent / "static" / "basemap_selector.js"
+
+    # The list of basemap names to make available for selection.
+    basemaps = traitlets.List([]).tag(sync=True)
+
+    # The currently selected basemap value.
+    value = traitlets.Unicode("").tag(sync=True)
+
     def __init__(self, basemaps: List[str], value: str):
         """Creates a widget for selecting a basemap.
 
         Args:
             basemaps (list): The list of basemap names to make available for selection.
             value (str): The default value from basemaps to select.
         """
+        super().__init__()
         self.on_close = None
         self.on_basemap_changed = None
+        self.basemaps = basemaps
+        self.value = value
+        self._setup_event_listeners()
 
-        self._dropdown = ipywidgets.Dropdown(
-            options=list(basemaps),
-            value=value,
-            layout=ipywidgets.Layout(width="200px"),
-        )
-        self._dropdown.observe(self._on_dropdown_click, "value")
-
-        close_button = ipywidgets.Button(
-            icon="times",
-            tooltip="Close the basemap widget",
-            button_style="primary",
-            layout=ipywidgets.Layout(width="32px"),
-        )
-        close_button.on_click(self._on_close_click)
-
-        super().__init__([self._dropdown, close_button])
+    def _setup_event_listeners(self) -> None:
+        self.on_msg(self._handle_message_event)
 
-    def _on_dropdown_click(self, change: dict) -> None:
-        """Handles the dropdown value change event.
+    def _handle_message_event(
+        self, widget: ipywidgets.Widget, content: Dict[str, Any], buffers: List[Any]
+    ) -> None:
+        del widget, buffers  # Unused
+        if content.get("type") == "click":
+            msg_id = content.get("id", "")
+            if msg_id == "close":
+                self.cleanup()
```

Comment:
This is great, but just wanted to leave a note that this is different from how the LayerManager widget works. In the LayerManager, we call the `remove()` method on the map.

Both are valid! I don't necessarily think we need to be consistent. I'm not sure which approach is better.

Reply:
For sure, it turns out that the toolbar tools have their own routine which depends on the "cleanup" method defined on the control so that they can automatically be processed for adding and removing. It's also how the toolbar button knows to turn off when clicked.

https://github.com/gee-community/geemap/blob/53078c92ed4ff680a30a1d59e05168133910b9bf/geemap/toolbar.py#L4368